### PR TITLE
Controller specs

### PR DESCRIPTION
**Pull request by zunairkhan811**
- Add tests for controllers
- Fix linter errors

### CODE REVIEW

File: spec/requests/posts_spec.rb [hunk 1]
```diff
@@ -2,16 +2,24 @@
...
-      get '/posts/index'
+    it 'returns http success for Posts index page' do
+      get '/users/:user_id/posts'
       expect(response).to have_http_status(:success)
     end
+    it 'renders the index template for posts' do
+      get '/users/:user_id/posts'
+      expect(response.body).to include('Show posts related to user ID')
+    end
   end
```
Comment (ShadyShawkat) [inline spec/requests/posts_spec.rb]: - [x] Good job setting up the tests. Kindly add one more test to each of the tested actions to test for the correct rendered template. As per [this](https://github.com/microverseinc/curriculum-rails/blob/main/blog-app/projects/controller_specs_project.md#:~:text=If%20a%20correct%20template%20was%20rendered.) requirement.

**Review by ShadyShawkat — CHANGES_REQUESTED** [review]
## Hi @zunairkhan811, 

Good job so far! 👍🏻 
There are some issues that you still need to work on to go to the next project but you are almost there!

## To Highlight
- Correct use of GitFlow ✔️ 
- Descriptive commit messages ✔️ 
- Professional looking pull request ✔️ 

Image

## Required Changes ♻️
_Check the comments under the review._

## Optional suggestions
_Every comment with the [OPTIONAL] prefix is not crucial enough to stop the approval of this PR. However, I strongly recommend you to take them into account as they can make your code better._

Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me in your question so I can receive the notification.

**_Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise._**

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561) you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

File: spec/requests/posts_spec.rb (hunk 1 above)
Comment (zunairkhan811) [reply spec/requests/posts_spec.rb]: @ShadyShawkat Thanks bro! The changes asked by you has been done 😃

**Review by Reem-lab — APPROVED** [review]
# Hi @zunairkhan811,  👋🏻👋🏻

Your project is complete!  There is nothing else to say other than... it's time to merge it :shipit: 
Congratulations! 🎉 


## Highlights

- [ ]  Followed Gitflow 
- [ ]  No Linters error ✅
- [ ]  Kudos for working on the controllers tests🤩
- [ ]  Great job making the requested changes  🚀


## Optional suggestions

_Every comment with the [OPTIONAL] prefix won't stop the approval of this PR. However, I strongly recommend you to take them into account as they can make your code better. Some of them were simply missed by the previous reviewer and addressing them will really improve your application._

Cheers, and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me in your question so I can receive the notification.

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561)  you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

**Pull request merged**